docs: Add update documentation (#2418)
starship/starship: README.md

   ```sh
   curl -fsSL https://starship.rs/install.sh | bash
   ```

>    To update starship itself, rerunnning the script above will work.

Reviewer: ```suggestion
   To update starship itself, rerunning the script above will replace the current version, keeping configs untouched.
```

>    To update starship itself, rerunnning the script above will replace the current version, keeping configs untouched.

Reviewer 2: I'm not really a _linguist_ and hate to nitpick about the wording but this seems more in line with the surrounding text:
```suggestion
   To update the Starship itself, rerun the above script. It will replace the current version without touching Starship's configuration.
```
